x-pack/plugins/ml/public/ui_actions/open_vis_in_ml_action.tsx

```diff
@@ -6,17 +6,20 @@
  */
 
 import { i18n } from '@kbn/i18n';
-import type { Embeddable } from '@kbn/lens-plugin/public';
-import type { MapEmbeddable } from '@kbn/maps-plugin/public';
+import { type EmbeddableApiContext, apiHasType } from '@kbn/presentation-publishing';
 import type { UiActionsActionDefinition } from '@kbn/ui-actions-plugin/public';
+import { apiHasLensConfig } from '@kbn/lens-plugin/public';
+import { apiHasMapConfig } from '@kbn/maps-plugin/public';
+import type { ActionApi } from './types';
 import { MlCoreSetup } from '../plugin';
-import { isLensEmbeddable, isMapEmbeddable } from '../application/jobs/new_job/job_from_dashboard';
 
 export const CREATE_LENS_VIS_TO_ML_AD_JOB_ACTION = 'createMLADJobAction';
 
+export const isApiCompatible = (api: unknown | null): api is ActionApi => Boolean(apiHasType(api));
```

Comment:
nit: is the `Boolean` cast necessary here? `apiHasType` should also return boolean.